src/wp-includes/user.php

```diff
@@ -3012,19 +3012,57 @@ function retrieve_password( $user_login = null ) {
 	 */
 	$message = apply_filters( 'retrieve_password_message', $message, $key, $user_login, $user_data );
 
+	// Wrap the single notification email arguments in an array to pass them to the retrieve_password_notification_email filter.
+	$retrieve_password_notification_email = array(
+		'to'      => $user_email,
+		'subject' => $title,
+		'message' => $message,
+		'headers' => '',
+	);
+
+	/**
+	 * Filters the contents of the reset password notification email sent to the user.
+	 *
+	 * @since x.x.x
+	 *
+	 * @param array   $retrieve_password_notification_email {
+	 *     Used to build wp_mail().
+	 *
+	 *     @type string $to      The intended recipient - user email address.
+	 *     @type string $subject The subject of the email.
+	 *     @type string $message The body of the email.
+	 *     @type string $headers The headers of the email.
+	 * }
+	 * @param string  $key        The activation key.
+	 * @param string  $user_login The username for the user.
+	 * @param WP_User $user_data  WP_User object.
+	 */
+	$retrieve_password_notification_email = apply_filters( 'retrieve_password_notification_email', $retrieve_password_notification_email, $key, $user_login, $user_data );
+
 	if ( $switched_locale ) {
 		restore_previous_locale();
 	}
 
-	if ( $message && ! wp_mail( $user_email, wp_specialchars_decode( $title ), $message ) ) {
-		$errors->add(
-			'retrieve_password_email_failure',
-			sprintf(
-				/* translators: %s: Documentation URL. */
-				__( '<strong>Error</strong>: The email could not be sent. Your site may not be correctly configured to send emails. <a href="%s">Get support for resetting your password</a>.' ),
-				esc_url( __( 'https://wordpress.org/support/article/resetting-your-password/' ) )
-			)
+	if ( $message ) {
+
+		$wp_mail_result = wp_mail(
+			$retrieve_password_notification_email['to'],
+			wp_specialchars_decode( $retrieve_password_notification_email['subject'] ),
+			$retrieve_password_notification_email['message'],
+			$retrieve_password_notification_email['headers']
 		);
+
+		if ( ! $wp_mail_result ) {
+			$errors->add(
+				'retrieve_password_email_failure',
+				sprintf(
+					/* translators: %s: Documentation URL. */
+					__( '<strong>Error</strong>: The email could not be sent. Your site may not be correctly configured to send emails. <a href="%s">Get support for resetting your password</a>.' ),
+					esc_url( __( 'https://wordpress.org/support/article/resetting-your-password/' ) )
+				)
+			);
+		}
+
 		return $errors;
```

Comment:
This should be inside the above conditional, otherwise a both a successful will return `$errors` instead of `true`.

Reply:
Done!